bump-formulae/main.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'formula'
+require 'utils/pypi'
+
+module Homebrew
+  module_function
+
+  def print_command(*cmd)
+    puts "[command]#{cmd.join(' ').gsub("\n", ' ')}"
+  end
+
+  def brew(*args)
+    print_command ENV["HOMEBREW_BREW_FILE"], *args
+    safe_system ENV["HOMEBREW_BREW_FILE"], *args
+  end
+
+  def read_brew(*args)
+    print_command ENV["HOMEBREW_BREW_FILE"], *args
+    output = `#{ENV["HOMEBREW_BREW_FILE"]} #{args.join(' ')}`.chomp
+    odie output if $CHILD_STATUS.exitstatus != 0
+    output
+  end
+
+  # Get formulae
+  formulae = ENV['HOMEBREW_BUMP_FORMULAE']
+  formulae = formulae.split(/[ ,\n]/).reject(&:blank?)
+
+  # Define additional message
+  message = 'Automated bump'
+
+  # Get livecheck info
+  json = read_brew 'livecheck',
+                    '--formula',
+                    '--quiet',
+                    '--newer-only',
+                    '--json',
+                    *formulae
+  json = JSON.parse json
+
+  # Define error
+  err = nil
+
+  # Loop over livecheck info
```

Comment:
Would be nice to use `brew bump` rather than `brew livecheck` here to also get Repology output. This was also the original intent of that tool so it might be nice to move all this Ruby logic into that tool.

Reply:
Good call. I'll put that on my TODO.

Reply:
@dawidd6 Came to this PR to make the same comment and found I'd already made it 😅. Any thoughts/luck with this?

Reply:
Heh, didn't have luck to sit and actually do this yet. Maybe someone else would like to pick this up? I'm not doing much open-source development lately :/

Reply:
Judging by the command I just ran, `brew bump` is harder to parse and never finds a repology version anyway so there isn't any gain there.

Reply:
@SMillerDev Given this is the only reason that command really exists, I think it's worth fixing it 😅 

> never finds a repology version anyway

Can you elaborate on that?

Reply:
CC @EricFromCanada @samford @nandahkrishna  who I think has played with `brew bump` a bit.

Reply:
Repology data fetching will be fixed in #12652 but `brew bump` could arguably benefit from a `--json` option (like `brew livecheck`) if we want to use it in this action.

I have a finely-aged stash (circa 2020-09) where I was implementing a `--json` option (alongside a `--no-pull-requests` option, to avoid fetching PR information when it's not beneficial). I'll try to dust off the `--json` part and turn it into a PR when I get a chance, as I believe it only needed a bit of polish when I last left it.

Reply:
> Repology data fetching will be fixed in [Homebrew/brew#12652](https://github.com/Homebrew/brew/pull/12652) but `brew bump` could arguably benefit from a `--json` option (like `brew livecheck`) if we want to use it in this action.

@samford I'm torn here. `--json` would be nice but I actually think nicer still would be to move more of this PR's Ruby logic into `brew bump` with a e.g. `--open-pr` flag. That was the original intent of the `brew bump` command and makes it more widely accessible outside this action.

Reply:
https://github.com/Homebrew/brew/issues/5725 was my original write-up for this that may be of interest.

Reply:
Have opened https://github.com/Homebrew/brew/pull/12702 to simplify this all into `brew bump --open-pr`